app/src/org/commcare/models/AndroidSessionWrapper.java

```diff
@@ -185,14 +185,19 @@ private static boolean ssdHasValidFormRecordId(int ssdId,
                 formRecordStorage.getMetaDataFieldForRecord(correspondingFormRecordId, FormRecord.META_STATUS));
     }
 
-    public void setCurrentStateAsInterrupted(FormIndex formIndex) {
+    public void setCurrentStateAsInterrupted(FormIndex formIndex, boolean autoSaving) {
         if (sessionStateRecordId != -1) {
             SqlStorage<SessionStateDescriptor> sessionStorage =
                     CommCareApplication.instance().getUserStorage(SessionStateDescriptor.class);
             SessionStateDescriptor current = sessionStorage.read(sessionStateRecordId);
             InterruptedFormState interruptedFormState = new InterruptedFormState(current.getID(), formIndex);
             HiddenPreferences.setInterruptedSSD(current.getID());
-            HiddenPreferences.setInterruptedFormState(interruptedFormState);
+            if (autoSaving) {
+                HiddenPreferences.setInterruptedFormState(interruptedFormState);
+            } else {
+                HiddenPreferences.clearInterruptedFormState();
+            }
```

Comment:
IMO think we should not introduce any additional non-required control flows in a hotfix branch as I am afraid we will need to do a QA on this otherwise but will let you take a call on this depending on how well you have tested these changes.